src/components/calculator.tsx

```diff
@@ -1,42 +1,266 @@
+        >
+          8
+        </div>
+        <div
+          onClick={() => handleNumber(9)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          9
+        </div>
+        <div
+          onClick={() => handleSign("*")}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          x
+        </div>
+        <div
+          onClick={() => handleNumber(4)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          4
+        </div>
+        <div
+          onClick={() => handleNumber(5)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          5
+        </div>
+        <div
+          onClick={() => handleNumber(6)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          6
+        </div>
+        <div
+          onClick={() => handleSign("-")}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          -
+        </div>
+        <div
+          onClick={() => handleNumber(1)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          1
+        </div>
+        <div
+          onClick={() => handleNumber(2)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          2
+        </div>
+        <div
+          onClick={() => handleNumber(3)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          3
+        </div>
+        <div
+          onClick={() => handleSign("/")}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          /
+        </div>
+        <div
+          onClick={handleSignInversion}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          +/-
+        </div>
+        <div
+          onClick={() => handleNumber(0)}
+          className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
+        >
+          0
+        </div>
```

Comment:
You could probably do something like this to avoid duplications.

```ts
const buttons = [
  "7",
  "8",
  "9",
  "*",
  "4",
  "5",
  "6",
  "-",
  "1",
  "2",
  "3",
  "/",
  "+/-",
  ".",
  // ...etc
  // the items in the buttons array should match the order of how you
  // want the buttons to be rendered
];
const signs = ["*", "+", "-", "/", "="]; // add rest of signs in array
```

And the functions `handleSign` and `handleNumber` can be combined into one

```ts
const handleButtons = (value: string) => {
  // then you check if the value is present inside of the signs array
  // if true, you write your logic for handling the signs and any other
  // additional checks you might need.
  // else we know the value is a number then you write your logic to
  // handle the number
};
```

And since the buttons are in an array you can just map through the array
and render the buttons instead manually hardcoding it.

```tsx
{
  buttons.map((button) => (
    <div
      key={button}
      onClick={() => handleButton(button)}
      className="text-center bg-slate-100 active:shadow-sm active:text-sm px-3 py-3 rounded-md flex items-center justify-center shadow-md"
    >
      {button}
    </div>
  ));
}
```

I don't know if this makes sense? If you need any clarification let me know.

I'm about to take a nap, I'll clone the repo and run it locally on my end when I wake up.

Reply:
@ArinzeJeffrey-droid Yeah. I was thinking of making the div a button and making it a reusable component but I first wanted to get the logic out of the way and get your opinion first before refactoring it. The sign array idea is fantastic. I wish I thought of that.

Reply:
Nice, making the div a reusable button is a good idea, you can do that too.

Reply:
@Mayor-Rhymes Yes it's possible.
Let's say you're on the `calculator-logic` branch and you create a branch from it and call it `branch-created-from-calculator-logic`, you can open a pull request and point the target branch to `calculator-logic` and merge the contents of  `branch-created-from-calculator-logic` into `calculator-logic`.

Does it make sense?

Reply:
```
const handleButtons = (value: string) => {
  // then you check if the value is present inside of the signs array
  // if true, you write your logic for handling the signs and any other
  // additional checks you might need.
  // else we know the value is a number then you write your logic to
  // handle the number
};
```
@ArinzeJeffrey-droid. As for the sign array, I am not sure if it is necessary. I particularly made the calculator screen a div instead of a text-input in order to prevent manually adding foreign values that would evaluate to an error. Furthermore, I am using a math library for the calculations and I trust it can handle evaluation errors without problems.